### PR TITLE
Conglomeration of pull requests 69-72

### PR DESCRIPTION
**Pull request by kellymclaughlin**
This is a giant mashup of 4 pull requests where one builds upon the changes from the previous. 
I have closed the other pull requests, but they can still be accessed to read the overview and see any relevant testing information. 

[69](https://github.com/basho/riak_moss/pull/69) - Add the notion of an administrative user
[70](https://github.com/basho/riak_moss/pull/70) - Integrate bucket serialization
[71](https://github.com/basho/riak_moss/pull/71) - Change user record structure and lay ground work for ACLs
[72](https://github.com/basho/riak_moss/pull/72) - Integrate user serialization

### CODE REVIEW

**Comment by jaredmorrow** [comment]
The last build of this branch didn't pass compile: http://buildbot-master.den.basho:8010/builders/test-moss/builds/83

**Comment by jaredmorrow** [comment]
Fixed: http://buildbot-master.den.basho:8010/builders/test-moss/builds/85

File: src/riak_moss_utils.erl [hunk 1]
```diff
@@ -66,73 +73,100 @@ close_riak_connection(Pid) ->
...
+                            {error, riak_moss_s3_response:error_code_to_atom(ErrorCode)};
+                        {error, _} ->
+                            CreateResult
+                    end;
+                {error, Reason1} ->
+                    {error, {riak_connect_failed, Reason1}}
+            end;
         {error, Reason} ->
-            {error, {riak_connect_failed, Reason}}
+            {error, Reason}
```
Comment (reiddraper) [inline src/riak_moss_utils.erl]: No need changing this, but figured I'd share a cool trick I learned for returning the same value as a Pattern.

``` erlang
{error, _Reason}=Error ->
    Error
```

File: src/riak_moss_utils.erl [hunk 2]
```diff
@@ -66,73 +73,100 @@ close_riak_connection(Pid) ->
...
     end.
 
 %% @doc Delete a bucket
 -spec delete_bucket(string(), binary()) -> ok.
-delete_bucket(KeyID, BucketName) ->
+delete_bucket(KeyId, Bucket) ->
     case riak_connection() of
         {ok, RiakPid} ->
             %% @TODO This will need to be updated once globally
```
Comment (reiddraper) [inline src/riak_moss_utils.erl]: does this TODO still apply?

Comment (kellymclaughlin) [inline src/riak_moss_utils.erl]: Nope, i'll remove it.

File: src/riak_moss_utils.erl [hunk 3]
```diff
@@ -66,73 +73,98 @@ close_riak_connection(Pid) ->
...
+                                           StanchionPort,
+                                           "application/json",
+                                           UserDoc,
+                                           [{ssl, StanchionSSL},
+                                            {auth_creds, AdminCreds}]),
+                    case CreateResult of
+                        ok ->
+                            {ok, User};
+                        {error, {error_status, _, _, ErrorDoc}} ->
+                            ErrorCode = xml_error_code(ErrorDoc),
```
Comment (reiddraper) [inline src/riak_moss_utils.erl:124]: Just putting a reminder for myself about the lager warning here. Do you think it's worth inspecting the actual ErrorDoc to see if the error is something
related to stanchion auth? Or just logging it no matter what?

**Comment by reiddraper** [comment]
merge it!

**Pull request closed without merging**